packages/xo-web/src/xo-app/vm/tab-network.js

```diff
@@ -71,22 +73,45 @@ import {
   subscribeResourceSets,
 } from 'xo'
 
+@addSubscriptions(props => ({
+  resourceSet: cb =>
+    subscribeResourceSets(resourceSets =>
+      cb(find(resourceSets, { id: props.resourceSet }))
+    ),
+}))
+@connectStore((state, props) => ({
+  isAdmin: isAdmin(state, props),
+  resolvedResourceSet: getResolvedResourceSet(
+    state,
+    props,
+    props.resourceSet !== undefined // to get objects as a self user
+  ),
+}))
 class VifNetwork extends BaseComponent {
   _getNetworkPredicate = createSelector(
     () => this.props.vif.$pool,
     vifPoolId => network => network.$pool === vifPoolId
   )
 
-  render() {
-    const { network } = this.props
+  _onChangeNetwork = network => {
+    const { resolvedResourceSet, vif } = this.props
+    return setVif(vif, {
+      network,
+      resourceSet: get(() => resolvedResourceSet.id),
```

Comment:
You can pass `props.resourceSet` instead.

Reply:
It's same thing because we need to pass the resourceSet ID and the `props.resourceSet`  will be an object or `undefined`.

Reply:
Right, I thought `resourceSet` was still an ID there.